include/ada/encoding_type.h

```diff
@@ -25,7 +25,7 @@ enum class encoding_type {
 /**
  * Convert a encoding_type to string.
  */
-ada_warn_unused std::string to_string(encoding_type type);
+ada_warn_unused const std::string to_string(encoding_type type);
```

Comment:
This changes the API, and if we do change the API, we might as well return a std::string_view instance. This will improve the efficiency of the code.

Note that there is nothing wrong with the existing code, except for the fact that it creates a new instance of an std::string which is wasteful (which the new code is likely to do as well).